sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -526,13 +526,13 @@ public static ResourceIdentifier Parse(string input)
         /// If the method returns false, result will be null.
         /// </param>
         /// <returns> True if the parse operation was successful; otherwise, false. </returns>
-        public static bool TryParse(string input, out ResourceIdentifier? result)
+        public static bool TryParse(string? input, out ResourceIdentifier? result)
         {
             result = null;
             if (string.IsNullOrEmpty(input))
                 return false;
 
-            result = new ResourceIdentifier(input);
+            result = new ResourceIdentifier(input!);
```

Comment:
Can you add a test case that passes in `string.Empty` and see what happens here?

Reply:
Like this one?

https://github.com/Azure/azure-sdk-for-net/blob/57e28159c7bfa3cb6d60bde6cd0448c8b53786eb/sdk/core/Azure.Core/tests/ResourceIdentifierTests.cs#L827-L833

Reply:
doh sorry was reading the diff wrong yes that one :-)